filters/builtin/inlinecontent.go

```diff
@@ -36,51 +37,40 @@ func NewInlineContent() filters.Spec {
 
 func (c *inlineContent) Name() string { return InlineContentName }
 
-func stringArg(a interface{}) (s string, err error) {
-	var ok bool
-	s, ok = a.(string)
-	if !ok {
-		err = filters.ErrInvalidFilterParameters
-	}
-
-	return
-}
-
 func (c *inlineContent) CreateFilter(args []interface{}) (filters.Filter, error) {
 	if len(args) == 0 || len(args) > 2 {
 		return nil, filters.ErrInvalidFilterParameters
 	}
 
-	var (
-		f   inlineContent
-		err error
-	)
+	var f inlineContent
 
-	f.text, err = stringArg(args[0])
-	if err != nil {
-		return nil, err
+	text, ok := args[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
 	}
+	f.template = eskip.NewTemplate(text)
 
 	if len(args) == 2 {
-		f.mime, err = stringArg(args[1])
-		if err != nil {
-			return nil, err
+		f.mime, ok = args[1].(string)
+		if !ok {
+			return nil, filters.ErrInvalidFilterParameters
 		}
 	} else {
-		f.mime = http.DetectContentType([]byte(f.text))
+		f.mime = http.DetectContentType([]byte(text))
```

Comment:
just nitpicking: can there be any special cases when the content type should be detected after the template has been resolved?

Reply:
The main motivation is to allow request/path properties to be reflected back in the response.
If there is enough hint in the template the content type would be properly detected, e.g.:
```
$ bin/skipper -inline-routes='* -> inlineContent("<html>${request.query.content}</html>") -> <shunt>'
$ bin/skipper -inline-routes='* -> inlineContent("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<response>${request.query.content}</response>") -> <shunt>'
```

Of course we can imagine something like
```
* -> inlineContent("${request.query.content}") -> <shunt>

```
and
```
curl "localhost:9090/?content=hello+world"
curl -v "localhost:9090/?content=<html>hello</html>"
```
In such cases content would not be detected. 
For such cases is content type is known upfront and does not change users may explicitly set it.

We can make content type to support templates as well then client could pass the value as another request property.
Or we can move content detection into runtime after template evaluation.

I would keep it as is and leave for future improvement if we ever find a usecase for content type detection.

Let me add tests for content type detection with templates though.

Reply:
Fair enough :+1:

Reply:
I hope nobody will use the examples in a production service, because it's "html injection as a sevice" ;)